api/preprint_providers/views.py

```diff
@@ -0,0 +1,189 @@
+from rest_framework import generics
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from website.models import Node, PreprintProvider
+
+from api.base import permissions as base_permissions
+from api.base.filters import ODMFilterMixin
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_object_or_error
+from api.base.pagination import MaxSizePagination
+
+from api.preprint_providers.serializers import PreprintProviderSerializer
+from api.preprints.serializers import PreprintSerializer
+
+
+class PreprintProviderMixin(object):
+    institution_lookup_url_kwarg = 'institution_id'
+
+    def get_preprintprovider(self):
+        provider = get_object_or_error(
+            PreprintProvider,
+            self.kwargs[self.institution_lookup_url_kwarg],
+            display_name='institution'
+        )
+        return provider
+
+
+class PreprintProviderList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin):
+    """
+    Paginated list of verified PreprintProviders available
+
+    ##PreprintProvider Attributes
+
+    OSF Institutions have the "institutions" `type`.
+
+        name           type               description
+        =========================================================================
+        name           string             title of the preprint provider
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the preprint provider's static logo
+        banner_path    string             a path to the preprint provider's static banner
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ]
```

Comment:
This read scope seems incorrect. Others use `NODE_BASE_READ` or `NODE_FILE_READ`, but I'm not certain that either of those are correct either. If we want to restrict read access to these views, consider adding a new read scope, e.g. `PREPRINT_READ`. If there isn't a reason to, `CoreScopes.ALWAYS_PUBLIC` is probably the right scope. I don't know why we would want to not tell people what preprint providers we have, but there might be a good reason.